src/jvm/main/org/jetbrains/kotlinx/lincheck/paramgen/strategy/real/RandomDoubleGenStrategy.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2023 JetBrains s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Lesser Public License for more details.
+ *
+ * You should have received a copy of the GNU General Lesser Public
+ * License along with this program.  If not, see
+ * <http://www.gnu.org/licenses/lgpl-3.0.html>
+ */
+
+package org.jetbrains.kotlinx.lincheck.paramgen.strategy.real;
+
+
+public interface RandomDoubleGenStrategy {
+
+    double nextDouble();
```

Comment:
All generators should use the same generation scheme you are introducing. Users should still be able to configure the range, but it should not affect the scheme.